apps/api/src/app/groups/groups.controller.ts

```diff
@@ -85,8 +85,28 @@ export class GroupsController {
     isGroupMember(
         @Param("id") groupId: string,
         @Param("member") member: string
-    ): boolean {
-        return this.groupsService.isGroupMember(groupId, member)
+    ) {
+        const isGroupMember = this.groupsService.isGroupMember(groupId, member)
+
+        let proof
+
+        if (isGroupMember) {
+            const _proof = this.groupsService.generateMerkleProof(
+                groupId,
+                member
+            )
+            proof = {
+                root: _proof.root.toString(),
+                leaf: _proof.leaf.toString(),
+                siblings: _proof.siblings.map((sibling) => sibling.toString()),
+                pathIndices: _proof.pathIndices
+            }
+        }
+
+        return {
+            isGroupMember,
+            proof
+        }
```

Comment:
```suggestion
        const { root, leaf, siblings, pathIndices } = this.groupsService.generateMerkleProof(
            groupId,
            member
        )
        return {
            root: root.toString(),
            leaf: leaf.toString(),
            siblings: siblings.map((sibling) => sibling.toString()),
            pathIndices: pathIndices
        }
```

What about creating two APIs? One to check if a member is part of a group, and another one to generate a proof

Reply:
That was my initial thought. 
But the route that made sense was `groups/:groupId/:member` which was already in use for checking membership. So thought of using the same route for `getMember` which would return both boolean and the proof (which is their membership details)
It could be two routes though - any suggestions?

Reply:
Got it, what about:

- `groups/:groupId/:member`: to check if a member is part of a group
- `groups/:groupId/:member/proof`: to get a membership proof of that member

Reply:
Ok, I will update.

Reply:
That api was already there :-D